realm/realm-library/src/androidTest/java/io/realm/RealmTests.java

```diff
@@ -1056,6 +1057,44 @@ public void compactRealm_onExternalStorage() {
         Realm.deleteRealm(config);
     }
 
+    private Pair<Long, Long> populateTestRealmAndCompactOnLaunch(CompactOnLaunchCallback compactOnLaunch) {
+        final String REALM_NAME = "test.realm";
+        RealmConfiguration realmConfig = configFactory.createConfiguration(REALM_NAME);
+        Realm realm = Realm.getInstance(realmConfig);
+        populateTestRealm(realm, 100);
+        realm.close();
+        long before = new File(realmConfig.getPath()).length();
+        realmConfig = configFactory.createConfiguration(REALM_NAME, compactOnLaunch);
+        realm = Realm.getInstance(realmConfig);
+        realm.close();
+        long after = new File(realmConfig.getPath()).length();
+        return new Pair(before, after);
+    }
+
+    @Test
+    public void compactOnLaunch_shouldCompact() throws IOException {
+        Pair<Long, Long> results = populateTestRealmAndCompactOnLaunch(new CompactOnLaunchCallback() {
+            @Override
+            public boolean shouldCompact(long totalBytes, long usedBytes) {
+                assertTrue(totalBytes > usedBytes);
+                return true;
+            }
+        });
+        assertTrue(results.first.compareTo(results.second) > 0);
```

Comment:
`Long > Long` works the same way `long > long` due to auto-unboxing https://docs.oracle.com/javase/tutorial/java/data/autoboxing.html